nats/src/services.rs

```diff
@@ -0,0 +1,30 @@
+use std::{collections::HashMap, sync::Arc};
+
+use bytes::Bytes;
+use lazy_static::lazy_static;
+use tokio::sync::{oneshot, RwLock};
+use tracing::warn;
+
+lazy_static! {
+    static ref REQUEST_WAITERS: Arc<RwLock<HashMap<String, oneshot::Sender<Bytes>>>> =
+        Arc::new(RwLock::new(HashMap::new()));
+}
+
+pub(crate) async fn add_request_waiter(subject: &str, sender: oneshot::Sender<Bytes>) {
+    let mut waiters = REQUEST_WAITERS.write().await;
+    waiters.insert(subject.to_string(), sender);
+}
+
+pub(crate) async fn dispatch_request_waiter(subject: &str, bytes: Bytes) {
+    let mut waiters = REQUEST_WAITERS.write().await;
+    if let Some(sender) = waiters.remove(subject) {
+        if let Err(_) = sender.send(bytes) {
+            warn!("Receiver side of request waiter dropped");
+        }
+    }
+}
```

Comment:
Is the request_waiter pattern a new recommended pattern in NATS to respond to incoming requests in a work-limited queue? I think I understand what's going on, it just seems a little complicated to deal with this lazy static to send responses

Reply:
This was a bit of an annoying aspect of implementing this. Keeping track of the waiters is basically required to correlate the replies coming from the actor with pending requests so that the NATS rust client can then use the `Respond` function.

If the stats/services API thing did this at a lower level, we wouldn't have to maintain this lookup table here.

Reply:
on the other side of the `sender.send` is a waiter getting ready to call `request.Respond`